File: dotnet/Lexy/Lexy.Compiler/Parser/Tokens/TokenizeResult.cs
using System;
using Lexy.Compiler.Language;
using Lexy.Compiler.Language.Expressions;

namespace Lexy.Compiler.Parser;

public class TokenizeResult : ParseResult<TokenList>
{
    public SourceReference Reference { get; }

    private TokenizeResult(TokenList result) : base(result)
    {
    }

    private TokenizeResult(bool success, SourceReference sourceReference, string errorMessage) : base(success, errorMessage)
    {
        Reference = sourceReference;
    }

    public static TokenizeResult Success(TokenList result)
    {
        if (result == null) throw new ArgumentNullException(nameof(result));

        return new TokenizeResult(result);
    }

    public static TokenizeResult Failed(SourceReference reference, string errorMessage)
    {
        return new TokenizeResult(false, reference, errorMessage);
    }
}

File: typescipt/existing/src/parser/tokens/tokenizeResult.ts


export class TokenizeResult extends ParseResult<TokenList> {
   public SourceReference Reference

   private TokenizeResult(TokenList result) : base(result) {
   }

   private TokenizeResult(boolean success, SourceReference sourceReference, string errorMessage) : base(success, errorMessage) {
     Reference = sourceReference;
   }

   public static success(result: TokenList): TokenizeResult {
     if (result == null) throw new Error(nameof(result));

     return new TokenizeResult(result);
   }

   public static failed(reference: SourceReference, errorMessage: string): TokenizeResult {
     return new TokenizeResult(false, reference, errorMessage);
   }
}
